pkg/scheduler/service/service.go

```diff
@@ -473,14 +473,25 @@ func (s *SchedulerServer) awaitFailover(schedulerProcess schedulerProcessInterfa
 
 func validateLeadershipTransition(desired, current string) {
 	log.Infof("validating leadership transition")
-	d := uid.Parse(desired).Group()
-	c := uid.Parse(current).Group()
-	if d == 0 {
+	desiredUID, err := uid.Parse(desired)
+	if err != nil {
+		log.Fatalf("Failed to parse desired scheduler UID %q: %v", desired, err)
+	}
+
+	desiredGroup := desiredUID.Group()
+	if desiredGroup == 0 {
 		// should *never* happen, but..
 		log.Fatalf("illegal scheduler UID: %q", desired)
 	}
-	if d != c && c != 0 {
-		log.Fatalf("desired scheduler group (%x) != current scheduler group (%x)", d, c)
+
+	currentUID, err := uid.Parse(desired)
```

Comment:
The current group would only be 0 if the current id is "", right? The 0 was coming from the default value of Group() from a nil (unparsable) UID.